src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -0,0 +1,181 @@
+/* Copyright (C) 2015-2022, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "opBuilderKVDB.hpp"
+
+#include <string>
+
+#include <fmt/format.h>
+#include <utils/stringUtils.hpp>
+#include <kvdb/kvdbManager.hpp>
+
+#include "syntax.hpp"
+
+
+namespace builder::internals::builders
+{
+
+using builder::internals::syntax::REFERENCE_ANCHOR;
+
+// <field>: +kvdb_extract/<DB>/<ref_key>
+types::Lifter opBuilderKVDBExtract(const types::DocumentValue & def)
+{
+    // Get target of the extraction
+    std::string target {json::formatJsonPath(def.MemberBegin()->name.GetString())};
+
+    // Get the raw value of parameter
+    if (!def.MemberBegin()->value.IsString())
+    {
+        throw std::runtime_error(
+            "Invalid parameter type for kvdb extracting operator (str expected)");
+    }
+
+    // Parse parameters
+    std::string params {def.MemberBegin()->value.GetString()};
+    auto parametersArr {utils::string::split(params, '/')};
+    if (parametersArr.size() != 3) //TODO: We are using default column only now
+    {
+        throw std::runtime_error("Invalid number of parameters for kvdb extracting operator");
+    }
+
+    // Get DB
+    KVDBManager& kvdbManager = KVDBManager::get();
+    KVDB& kvdb = kvdbManager.getDB(parametersArr[1]);
+    if (kvdb.getState() != KVDB::State::Open)
```

Comment:
i think this should rather be a member function of kvdb like `kvdb.isReady()`

Reply:
I like it!
Changed.